x-pack/plugins/alerting/server/authorization/alerting_authorization_kuery.ts

```diff
@@ -35,26 +37,40 @@ const esQueryConfig: EsQueryConfig = {
 
 export function asFiltersByRuleTypeAndConsumer(
   ruleTypes: Set<RegistryAlertTypeWithAuth>,
-  opts: AlertingAuthorizationFilterOpts
+  opts: AlertingAuthorizationFilterOpts,
+  spaceId: string | undefined
 ): KueryNode | JsonObject {
   const kueryNode = nodeBuilder.or(
     Array.from(ruleTypes).reduce<KueryNode[]>((filters, { id, authorizedConsumers }) => {
       ensureFieldIsSafeForQuery('ruleTypeId', id);
-      filters.push(
-        nodeBuilder.and([
-          nodeBuilder.is(opts.fieldNames.ruleTypeId, id),
-          nodeBuilder.or(
-            Object.keys(authorizedConsumers).map((consumer) => {
-              ensureFieldIsSafeForQuery('consumer', consumer);
-              return nodeBuilder.is(opts.fieldNames.consumer, consumer);
-            })
-          ),
-        ])
-      );
+      const andNodes = [
+        nodeBuilder.is(opts.fieldNames.ruleTypeId, id),
+        nodeBuilder.or(
+          Object.keys(authorizedConsumers).map((consumer) => {
+            ensureFieldIsSafeForQuery('consumer', consumer);
+            return nodeBuilder.is(opts.fieldNames.consumer, consumer);
+          })
+        ),
+      ];
+
+      if (opts.includeSpaceId && opts.fieldNames.spaceIds != null && spaceId != null) {
+        andNodes.push(nodeBuilder.is(opts.fieldNames.spaceIds, spaceId));
+      }
+
+      filters.push(nodeBuilder.and(andNodes));
+
       return filters;
     }, [])
   );
 
+  if (ruleTypes.size === 0 && opts.includeSpaceId && spaceId != null) {
```

Comment:
Can we create a new function in the alerting authorization kuery class called `asFiltersBySpaceId`? and call that instead of adding this exception case inside `asFiltersByRuleTypeAndConsumer`. I think it is unexpected behavior that passing no rule types will actually give access to all rule types as long as space id is defined.

Reply:
Updated here https://github.com/elastic/kibana/pull/105333/commits/4f8972a0666b3e7a325939b64b77546540b31b06

Thanks!